content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -0,0 +1,258 @@
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: Restrict)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `Restrict`?
+
+If a `User` has a related post in the `Post` table you will **not** be able to delete the `User`, and you will receive a runtime error. You will be able to delete (all) the `Post`s however.
+
+<Admonition type="warning">
+
+The `Restrict` action is not is **not** available on [SQL Server](../../../database-connectors/microsoft-sql-server) and will trigger a schema validation error. Instead, you can use [`NoAction`](#noaction), which produces the same result.
+
+</Admonition>
+
+### NoAction
+
+The `NoAction` action is similar to `Restrict` in that it will prevent the update or deletion of any records which are connected and throw an error. The difference between the two is dependant on the database being used. But in general they both have the same functionality. 
+
+In Postgres however, `NoAction` defers the check until later in the transaction.
+
+TODO::: add SQLSERVER, MYSQL
+
+#### Example usage
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: NoAction)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `NoAction`?
+
+If a `User` has a related post in the `Post` table you will **not** be able to delete them. This will result in a runtime error.
+
+### SetNull
+
+The `SetNull` action will set relation [scalar fields](../data-model#scalar-fields) to null if the relation is updated or deleted. If one or more of the relations scalar fields are required a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: SetNull)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `SetNull`?
+
+If a `User` is deleted then the corresponding `authorId` field in the `Post` table will be set to `NULL`. It will be left in the database, floating there without a connection to any `User`.
+
+### SetDefault
+
+The `SetDefault` action will set relation [scalar fields](./#annotated-relation-fields-and-relation-scalar-fields) to their default values upon updated or deletion of a related record. If no defaults are provided for any of the scalar fields, a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: SetDefault)
+  authorId  Int
```

Comment:
I think for the example to work, you would need to add `@default` to the `authorId` field. It needs to be a database-level default, Prisma-level defaults won't work here.

In my example I used usernames of type String as IDs and set the default of the scalar field to `"anonymous"`. Using integers might be less illustrative but you could then use something like `@default(1)` and add a note that a user with ID one needs to exist.